src/cargo/ops/common_for_install_and_uninstall.rs

```diff
@@ -542,66 +542,12 @@ where
     }
 
     if let Some(name) = name {
-        let vers = if let Some(v) = vers {
-            // If the version begins with character <, >, =, ^, ~ parse it as a
-            // version range, otherwise parse it as a specific version
-            let first = v
-                .chars()
-                .next()
-                .ok_or_else(|| format_err!("no version provided for the `--vers` flag"))?;
-
-            let is_req = "<>=^~".contains(first) || v.contains('*');
-            if is_req {
-                match v.parse::<VersionReq>() {
-                    Ok(v) => Some(v.to_string()),
-                    Err(_) => bail!(
-                        "the `--vers` provided, `{}`, is \
-                         not a valid semver version requirement\n\n\
-                         Please have a look at \
-                         https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html \
-                         for the correct format",
-                        v
-                    ),
-                }
-            } else {
-                match v.to_semver() {
-                    Ok(v) => Some(format!("={}", v)),
-                    Err(e) => {
-                        let mut msg = format!(
-                            "the `--vers` provided, `{}`, is \
-                             not a valid semver version: {}\n",
-                            v, e
-                        );
-
-                        // If it is not a valid version but it is a valid version
-                        // requirement, add a note to the warning
-                        if v.parse::<VersionReq>().is_ok() {
-                            msg.push_str(&format!(
-                                "\nif you want to specify semver range, \
-                                 add an explicit qualifier, like ^{}",
-                                v
-                            ));
-                        }
-                        bail!(msg);
-                    }
-                }
-            }
-        } else {
-            None
-        };
-        let vers = vers.as_deref();
-        let vers_spec = if vers.is_none() && source.source_id().is_registry() {
-            // Avoid pre-release versions from crate.io
-            // unless explicitly asked for
-            Some("*")
-        } else {
-            vers
-        };
-        let dep = Dependency::parse_no_deprecated(name, vers_spec, source.source_id())?;
+        let vers = vers.map(|v| v.to_string());
```

Comment:
I'm a bit uncertain, why take a string, parse it to a VersionReq, only to immediately convert it back to a string where `parse_no_deprecated` will parse it again?  Couldn't `select_pkg` keep the `Option<&str>` to avoid that?